src/routes/categories/containers/CategoriesPage.js

```diff
@@ -73,35 +102,55 @@ class CategoriesPage extends React.Component {
   }
 
   render () {
-    const url = this.getParentPath()
-    const hierarchy = new Hierarchy(this.props.path)
+    const category = CategoryModel.getCategoryByPath(this.props.categories, this.props.path)
+    if (!category) {
+      // todo show an error that the page has not been found, redirect to the notFound route
+    }
+    const children = this.props.categories.filter(_category => category.children.includes(_category.id))
 
-    // Pass data to hierarchy
-    const error = hierarchy.build(this.props.pages)
-    if (error) {
-      return <Error error={error} />
+    let Content
+    if (category.children.length === 0) {
+      // last level, our category is a simple page
+      Content = <Page page={category} />
+    } else if (category.id === 0) {
+      // first level, we want to display a table with all first order categories
+      Content = <CategoryTiles categories={children}
+                               title={category.title}
+                               locations={this.props.locations} />
+    } else {
+      // some level between, we want to display a list
+      Content = <TitledContentList categories={children}
+                                   parentCategory={category} />
     }
 
     return <div>
       <Breadcrumb
-        hierarchy={hierarchy}
-        language={this.props.language}
-        location={this.props.location}
-      />
-      <Content url={url} hierarchy={hierarchy} />
+        categories={this.props.categories}
+        category={category}
+        locations={this.props.locations} />
+      {Content}
       <PdfButton href={this.getPdfFetchPath()} />
     </div>
   }
 }
 
+const mapDispatchToProps = (dispatch) => ({
+  dispatchLanguageChangeUrls: (urls, languages, availableLanguages) => dispatch(
+    setLanguageChangeUrls(urls, languages, availableLanguages)
+  ),
+  dispatchRedirectToUrl: (url) => dispatch(replace(url))
+})
+
 const mapStateToProps = (state) => ({
   language: state.router.params.language,
   location: state.router.params.location,
-  path: state.router.params['_'] // _ contains all the values from *
+  path: state.router.params['_'], // _ contains all the values from *
+  categoryId: state.router.query.id
```

Comment:
imo not, it's not that much logic for the redirect, and I find this more logic because otherwise we dispatch the languageChangeUrls in CategoriesPage, go to the redirectPage on an actual language change, and redirect back to CategoriesPage, which isn't quite logic for me.
@maxammann also said that we want to get rid of it.

but if you both agree it would be better to still have this RedirectPage, i can refactor this of course

Reply:
I think the same, but the redirect pages will be gone as soon as my task is merged in the cms. So we can leave it for now.